src/Compiler/Checking/CheckExpressions.fs

```diff
@@ -5119,22 +5119,75 @@ and TcPatLongIdentActivePatternCase warnOnUpper (cenv: cenv) (env: TcEnv) vFlags
     let vExprTy = vExpr.Type
 
     let activePatArgsAsSynPats, patArg =
-        match args with
-        | [] -> [], SynPat.Const(SynConst.Unit, m)
-        | _ ->
-            // This bit of type-directed analysis ensures that parameterized partial active patterns returning unit do not need to take an argument
-            let dtys, retTy = stripFunTy g vExprTy
-
-            if dtys.Length = args.Length + 1 &&
-                ((isOptionTy g retTy && isUnitTy g (destOptionTy g retTy)) ||
-                (isValueOptionTy g retTy && isUnitTy g (destValueOptionTy g retTy))) ||
-                // `bool` partial AP always be treated as `unit option`
-                // For `val (|P|_|) : _ -> bool`, only allow `match x with | P -> ...`
-                // For `val (|P|_|) : _ -> _ -> bool`, only allow `match x with | P parameter -> ...`
-                (not apinfo.IsTotal && isBoolTy g retTy) then
+        let rec IsNotSolved ty =
+            match ty with
```

Comment:
Shall `ty` be stripped here? @T-Gro

Reply:
Stripping would remove solved TType_var.
I think for the case of resolving solved/unsolved typars, this is good.

What is not covered and stripping would do:
- Aliases - I cannot come up with a way of placing an alias here
- UoMs - not relevant for IsNotSolved
- ByRefs - not relevant for IsNotSolved